### DISCUSSION TITLE
Parquet Hive partitioning , Partition by range in numbers

### DISCUSSION THREAD

**Discussion by MPizzotti** [duckdb/duckdb#11691] [Q&A]
Often, while working on a dataset that cointains a integer key, when exporting to parquet for later use, i found out that querying back the data on a specific ID was quite slow, since the Parquet had to scan the entire column + row group.

Since Hive Partitioning is already available for running a filter pushdown on Dates,

I was wondering if is possibile to Hive partion By a fixed range on numbers.

Example:
i have a dataset with aprox. 150million rows but with 18k IDs (let's say a timeseries-like dataset).
i want to hive partion the dataset in groups of 1k IDs:
```
root_folder
    |_hive_key_id=[1-1000]
        |_file_1.parquet
        |_file_2.parquet
        |_...
    |_hive_key_id=[1001-2000]
        |_file_1.parquet
        |_file_2.parquet
        |_...
    |_hive_key_id=[2001-3000]
        |_file_1.parquet
        |_file_2.parquet
        |_...
    |_..
    |_hive_key_id=[17001-18000]
        |_file_1.parquet
        |_file_2.parquet
        |_...
```

**Comment by nbc** [comment]
Hi,

I'm quite sure partitionning by range is not possible but you can achieve almost the same goal by sorting the dataset by your ID when writing it.

[More information here](https://medium.com/@pankajroark/sorting-and-parquet-3a382893cde5)

**Comment by soerenwolfers** [comment]
Just add a `id_thousands = i // 1000` column to your data.

**Comment by SahilBijlani** [comment]
IS THIS ISSUE UNASSIGNED? CAN I WORK ON IT?

**Reply by soerenwolfers** [reply]
I'd say it's unlikely that there is appetite to have this in the codebase, since there is such an easy way to achieve this by hand.

**Reply by mitstake** [reply]
Was going to ask about this same feature just now before finding this thread. 

So if I want to partition my data by month, would the suggestion be to write this ideal query:

    select *
    from data
    where datetime between '2020-05-07' and '2021-04-13 23:59:59'

Like this instead:

    select *
    from data
    where yearmonth between '2020-05' and '2021-04' 
        and datetime between '2020-05-07' and '2021-04-13 23:59:59'

Where we would need to create a yearmonth hive-partition and reference it in the query in addition to our actual desired datetime filter?
